### DISCUSSION TITLE
Just migrated from 4.x to 5.x and need some help.

### DISCUSSION THREAD

**Discussion by suprsidr** [feathersjs/feathers#3536] [Q&A]
In 4.x when I sent a patch from the client I would get 2 responses:
```
[
    null,
    {
        ...userData
    }
],
[
    "users patched",
    {
        ...userData
    }
]
```

My client side event listeners fired as expected.
But now in 5.x I only get the first.
```
[
    null,
    {
        ...userData
    }
],
```

And my client side event listeners are not firing.

Anyone have any clue what I missed on the update?

**Comment by daffl** [comment]
You should be getting separate events for each individual entry in the array - but this should've already been the case in v4. Do you mean you are only getting an event for the first entry or did you get the full array back before?

**Reply by suprsidr** [reply]
I now get only one response in the websocket and it has `null` for the first entry.
Previously in v4 I was getting 2 responses and the first entry in the second response was "users patched".

And this is the user service which is almost entirely a feathers service.

in the client:
```
const service = client.service("users");

const result = await service.patch(id, values);
console.log(result);

```

elsewhere in the app:
```
const service = client.service("users");
service.on("patched", console.log); // this never fires but it did in v4

```

**Reply by daffl** [reply]
Indeed, I am seeing the same issue in the Feathers chat but I'm not sure why (all the individual pieces seem to have tests to ensure they work with arrays). Requires further investigation.
